docs/src/main/asciidoc/se/openapi/openapi-generator.adoc

```diff
@@ -23,9 +23,280 @@
 :rootdir: {docdir}/../..
 
 include::{rootdir}/includes/se.adoc[]
+:incdir: {rootdir}/includes/openapi
+:gen-inc: {incdir}/openapi-generator.adoc
+:snippets: {sourcedir}/se/openapi/OpenApiGeneratorSnippets.java
 
-== Coming Soon
+:helidon-client-xref: {webclient-page}
 
-The support for link:{openapi-generator-tool-site-url}[OpenAPI generator] is not available yet.
+include::{gen-inc}[tag=preamble]
 
-You can track the progress for this feature here: https://github.com/helidon-io/helidon/issues/7943.
+include::{gen-inc}[tags=intro;coords;config;usage-basics;usage-grouping-intro;usage-grouping-server]
+
+The generator creates Helidon routing logic based on the longest common path prefix shared among the operations that are grouped into each API.
+
+[NOTE]
+====
+If the operations in an API have no common prefix then the generated routing will be inefficient at runtime.
+The generator logs a warning and includes a `TODO` comment in the generated routing.
+
+Review the paths and the `tags` settings in your OpenAPI document and consider revising one or the other so all operations in each API share a common path prefix. If you do not have control over the OpenAPI document or do not want to change it, consider specifying the generator option `x-helidon-groupBy first-path-segment` which groups operations into APIs not by `tags` value but by the first segment of each operation's path.
+====
+
+include::{gen-inc}[tags=usage-grouping-client;usage-after-grouping;using-generated-code-intro;using-generated-code-server-intro]
+
+If you choose to generate interfaces for the APIs, the generator creates routing rules for the API services it generates but you write virtually all of the logic to process incoming requests by implementing the very short methods generated in the implementation class.
+
+The rest of this sections focuses on your next steps if, on the other hand, you decide to generate abstract classes.
+
+==== What you _must_ do: implement your business logic and send the response
+The generator creates an implementation class as well as the abstract class for each API.
+The implementation class contains a `handle` method for each API operation with a very simple method body that returns a not-yet-implemented HTTP status in the response. The following example shows the generated method for the `addPet` OpenAPI operation.
+[source,java]
+.The generated `handleAddPet` method in the `PetApiImpl` class
+----
+include::{snippets}/[tags=PetServiceImplCustomclass-header;PetServiceImplCustomhandleAddPetDecl;PetServiceImplCustomhandleAddPetGenerated;PetServiceImplCustomhandleAddPetEnd, indent=0]
+}
```

Comment:
Are these close curly expected ?

Reply:
Yes, they are needed but I added tags to the snippet to bring them in from there instead of having them _after_ the `include`.